### ISSUE TITLE
Azure Storage Explorer hangs at "Waiting for authentication"

### ISSUE THREAD

**Issue by deepix** [Microsoft/AzureStorageExplorer#126]
**Storage Explorer Version**: 1.0.0

**Platform**: macOS

**Architecture**: x64

**Build Number**: 20180415.2

**Commit**: 5d603052

**Regression From**: 0.9.6

#### Steps to Reproduce: ####

1. Open Azure Storage Explorer
2. Click on the human icon, click on "Sign In ..."

#### Expected Experience: ####
Show a dialog box to sign in to my Azure account.

#### Actual Experience: ####
Application hangs at "Waiting for authentication".

<img width="785" alt="screen shot 2018-05-02 at 10 51 34 am" src="https://user-images.githubusercontent.com/1882239/39540372-5067b5e0-4df7-11e8-80d5-acc46315f77c.png">

**Comment by dbiswas** [comment]
This issue happens because your keychain is open for some reason and not closed.
So, open the Mac Application: Key Chain Access.
And then select the lock icon and close it.
And now restart the Storage Explorer.

**Comment by alpeshpandya** [comment]
This still issue persists after locking keychain. The authentication screen gets stuck with "taking to your organization's login page" and then a blank screen for ever never prompting for the organization authentication. 

az login works similar way on my macos. It is only storage explorer that gets stuck.
<img width="1042" alt="Screen Shot 2019-08-10 at 12 46 46 PM" src="https://user-images.githubusercontent.com/2990104/62825188-64e3f080-bb6d-11e9-965c-88c1114801fb.png">

**Comment by voronoi** [comment]
> This still issue persists after locking keychain. The authentication screen gets stuck with "taking to your organization's login page" and then a blank screen for ever never prompting for the organization authentication.
> 
> az login works similar way on my macos. It is only storage explorer that gets stuck.
> <img alt="Screen Shot 2019-08-10 at 12 46 46 PM" width="1042" src="https://user-images.githubusercontent.com/2990104/62825188-64e3f080-bb6d-11e9-965c-88c1114801fb.png">

I am facing a similar issue as the user quoted. Locking or Unlocking the keychain seems to make no difference in my case.

**Comment by voronoi** [comment]
Still facing the same issue as above even after upgrading to Storage Explorer v1.11.1

**Comment by aleacosta5** [comment]
I'm having the same exact issue as @alpeshpandya.

I tried deleting the folders inside Users /<myUserName>/Library/Application Support/StorageExplorer

Also tried deleting the keyChain associated with Storage Explorer.

System Details:
macOS: 10.15.2
Storage Explorer v1.11.1 and also tried with v1.11.2

**Comment by Big0Bertha** [comment]
Same issue as @aleacosta5 here.

**Comment by joeyslack** [comment]
For anyone reading this, I solved this by simply opening keychain (and doing nothing), and re-authenticating.

Regardless, this should still be considered an issue by Microsoft, and issue should not be marked resolved.

Having to "fidget" with keychain worries me greatly. If anything, just update the app to say "Cannot authenticate due to keychain access, please open keychain to resume", or something.

**Comment by shary-sharath** [comment]
> Hey @deepix , can you try the following:
> 
> 1. Close Storage Explorer
> 2. Open keychain (cmd  + space, type in keychain, hit enter)
> 3. Select the "login" keychain
> 4. Click the padlock icon to lock the keychain (the padlock will animate to a locked position when complete, it may take a few seconds depending on what apps you have open)
>    ![image](https://user-images.githubusercontent.com/21112697/39140751-6f08d00c-46da-11e8-8f8f-14fd25d1f5fb.png)
> 5. Launch Storage Explorer
> 6. A pop up should appear saying something like "Service hub wants to access the keychain", enter your Mac admin account password and click **Always Allow** (or **Allow** if **Always Allow** is not available)
> 7. Try to sign in

I am getting this issue in Windows 10. How do I resolve this?

**Comment by knightian** [comment]
Given that the login keychain is unlocked on login...... how is this not fixed? In big sur I cannot even lock the login keychain... srsly?

**Comment by tadywankenobi** [comment]
Yup, same as @knightian here. Can't lock login from Keychain Access any more.

This needs a fix, it's been open for over 2 years...

**Comment by tadywankenobi** [comment]
Actually, I just realised I was running the application from the downloads folder. Moved it into Applications and at the same time, set the Keychain Access to lock after 1 minute and then waited a few. When I opened Storage Explorer this time, it asked me for my system password and connected to Azure. Still have permissions issues, but that looks like something else...
